packages/block-library/src/navigation-link/index.php

```diff
@@ -367,5 +386,42 @@ function register_block_core_navigation_link() {
 			'variations'      => array_merge( $built_ins, $variations ),
 		)
 	);
+
+	// Register actions for all post types and taxonomies not registered until now.
+	// This needs to happen in this function, because otherwise for post types/taxonomies
+	// registered before this block, those functions would add variations to a block that don't exist.
```

Comment:
```suggestion
	// Register actions for all post types and taxonomies that may not yet be registered.
	// This needs to happen in this function, because otherwise any post types/taxonomies
	// registered **before** this block will attempt to register on a block that does not yet exist.
```